### PR TITLE
Swap roles of xtval and xtval2

### PR DESCRIPTION
**Pull request by andresag01**
Swap the roles of xtval and xtval2 such that xtval holds the address on a CHERI fault and xtval2 holds the extra CHERI cause and type information for the exception. The following also had to change along the way:

* Update old references to xtval that must now point to xtval2
* Add `htval2` because otherwise we do not get any CHERI information in HS-mode
* Adjust diagrams and descriptions for xtval and xtval2

This PR is a follow up to https://github.com/riscv/riscv-cheri/pull/373

### CODE REVIEW

**Review by arichardson — COMMENTED** [review]
I agree this is more consistent. I do wonder if we want to allocate more values in *cause instead of adding all the *tval2 registers or if that is a more contentious change.

Maybe allocating bits 24-31 of mcause for a "sub-cause" field could be easier since mcause >= 64 is reserved?

But regardless of whether we want to make that change I think this makes it more consistent with other exceptions so LGTM.

File: src/hypervisor-integration.adoc [hunk 1]
```diff
@@ -169,30 +190,30 @@ include::img/vstdcreg.edn[]
...
 
-<<vstval2>> holds the same set of values that <<stval2>> can hold.
+NOTE: <<vstval2>> and <<stval2>> are not standard RISC-V CSRs, but <<mtval2>>
+is.
 
-.Virtual supervisor trap value register 2
-[#vstval2-format]
-include::img/vstval2reg.edn[]
+#.Virtual supervisor trap value register 2
+#[#vstval2-format]
```
[the remark this answers is not in the document]
Comment (andresag01) [reply src/hypervisor-integration.adoc:215]: Typo. I will fix it

**Comment by andresag01** [comment]
>I do wonder if we want to allocate more values in *cause instead of adding all the *tval2 registers or if that is a more contentious change.

We considered this before, but didn't take that approach because it would consume too much space in *cause, and other places connected to it like medeleg. In future, we will probably need to remove the CHERI fault cause 28 and instead use the new software-check fault cause (see https://github.com/riscv/riscv-cheri/issues/60).

IMHO, we should keep what we currently have in the spec (and this PR) and await feedback from the ARC.

**Comment by tariqkurd-repo** [comment]
> > I do wonder if we want to allocate more values in *cause instead of adding all the *tval2 registers or if that is a more contentious change.
> 
> We considered this before, but didn't take that approach because it would consume too much space in *cause, and other places connected to it like medeleg. In future, we will probably need to remove the CHERI fault cause 28 and instead use the new software-check fault cause (see #60).
> 
> IMHO, we should keep what we currently have in the spec (and this PR) and await feedback from the ARC.

I agree with @andresag01 , I think it's better to keep what we have and all this area is subject to change by the ARC

**Review by tariqkurd-repo — APPROVED** [review]

File: src/riscv-integration.adoc [hunk 2]
```diff
@@ -744,25 +744,51 @@ WARNING: *CHERI v9 Note:* Encoding and values changed, and generally were
...
+==== Machine Trap Value Register 2 (mtval2)
+
+ifdef::cheri_v9_annotations[]
+WARNING: *CHERI v9 Note:* This CSR was not part of CHERIv9.
+endif::[]
+
+The <<mtval2>> register is an MXLEN-bit read-write register, which is added as part of the
+Hypervisor extension cite:[riscv-priv-spec]. {cheri_base_ext_name} also requires the implementation of this CSR.
+
+When a CHERI fault is taken into M-mode, <<mtval2>> written
```
Comment (andresag01) [inline src/riscv-integration.adoc:774]:
```suggestion
When a CHERI fault is taken into M-mode, <<mtval2>> is written
```

**Pull request merged**